src/Sylius/Component/Core/Resolver/DefaultShippingMethodResolver.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Resolver;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Shipping\Model\ShipmentInterface;
```

Comment:
The problem is, it implements interface from *Shipping* component. There is no problem with calling ``getOrder()``, as we [assert that subject is shipment from Core](https://github.com/Sylius/Sylius/pull/5220/files#diff-8f588422e1d608e78fb3477db05afd6fR43), but we must keep types compatibility between this *Core* service and *Shipping* interface.